app/lib/features/chat/widgets/image_message_builder.dart

```diff
@@ -38,6 +38,7 @@ class _ImageMessageBuilderState extends State<ImageMessageBuilder> {
             .imageBinary(widget.message.id)
             .then((value) => value.asTypedList()),
       );
+      setState(() {});
```

Comment:
what is this for? maybe add a doc comment because it looks like something someone might not understand what it is for and thus just deletes it in the future..

Reply:
Yes sure, I will add doc comment over there.

This is related to `image message` loading issue on the chat messages.
Actually, there was `async` method called from` initState()` to load image data but prior to that `build()` call and initialise with no image and going for loading forever.

So i have added  `setState(() {});` to update the image view when image data are loaded in the variable.

```

class _ImageMessageBuilderState extends State<ImageMessageBuilder> {
  AsyncValue<Uint8List> imageData = const AsyncValue.loading();

  @override
  void initState() {
    super.initState();
    loadImage();
  }

  Future<void> loadImage() async {
    try {
      imageData = AsyncValue.data(
        await widget.convo
            .imageBinary(widget.message.id)
            .then((value) => value.asTypedList()),
      );
      setState(() {});
    } catch (e, s) {
      imageData = AsyncValue.error(e, s);
    }
  }

  @override
  Widget build(BuildContext context) {
    final size = MediaQuery.of(context).size;
    return imageData.when(
      data: (data) {
        return ClipRRect(
          borderRadius: widget.isReplyContent
              ? BorderRadius.circular(6)
              : BorderRadius.circular(15),
          child: ConstrainedBox(
            constraints: BoxConstraints(
              maxHeight:
                  widget.isReplyContent ? size.height * 0.2 : size.height * 0.3,
              maxWidth:
                  widget.isReplyContent ? size.width * 0.2 : size.width * 0.3,
            ),
            child: Image.memory(
              data,
              frameBuilder: (
                BuildContext context,
                Widget child,
                int? frame,
                bool wasSynchronouslyLoaded,
              ) {
                if (wasSynchronouslyLoaded) {
                  return child;
                }
                return AnimatedOpacity(
                  opacity: frame == null ? 0 : 1,
                  duration: const Duration(seconds: 1),
                  curve: Curves.easeOut,
                  child: child,
                );
              },
              errorBuilder: (
                BuildContext context,
                Object url,
                StackTrace? error,
              ) {
                return Text('Could not load image due to $error');
              },
              fit: BoxFit.cover,
            ),
          ),
        );
      },
      error: (e, st) => Text('Error loading image $e'),
      loading: () => const CircularProgressIndicator(),
    );
  }
}
```